js/spotify.js

```diff
@@ -1,12 +1,125 @@
 // API Docs at: 
 // https://developer.spotify.com/technologies/web-api/search/
 
+'use strict';
 
-function searchByArtist(keyword) {
-  var url = 'http://ws.spotify.com/search/1/artist.json?q='+keyword;
+// Save existing DOM elements as variables
+const $results = $('#results');
+const $searchKeyword = $('#search-keyword');
+const $search = $('#search');
+
+// Take user search term and determine whether to search it as track or artist
+const getKeyword = (e) => {
+  e.preventDefault();
+
+  // clear old results, if any
+  $results.empty();
+
+  var keyword = $searchKeyword.val();
+
+  if ($(':selected').val() === 'artist') {
+    composeQuery(keyword, 'artist');
+  } else {
+    composeQuery(keyword, 'track');
+  }
+};
+
+// Using search term and search type, compose the Spotify endpoint URL
+const composeQuery = (keyword, searchType) => {
+  var url = 'https://api.spotify.com/v1/search?q=' + keyword + '&type=' + searchType;
+
+  searchSpotify(url);
```

Comment:
Love the use of a `composeQuery` function. I think you have the right idea here. I would not include the `searchSpotify()` function call here though. We want our functions to focus on one task. By having `searchSpotify()` function call, we _assume_ that anytime we compose a query that we **immediately** want to search spotify which may or may not be the case!